.github/workflows/tests.yml

```
@@ -0,0 +1,43 @@
+name: "Tests"
+
+on:
+  - pull_request
+  - push
+
+jobs:
+  test:
+    name: PHP ${{ matrix.php }}
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        php:
+          - "5.3"
+          - "5.4"
+          - "5.5"
+          - "5.6"
+          - "7.0"
+          - "7.1"
+          - "7.2"
+          - "7.3"
+          - "7.4"
+          - "8.0"
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+
+      - name: Setup PHP ${{ matrix.php }}
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+
+      - name: Lint PHP source files
+        run: |
+          # Lint All PHP source files
+          # Remove source expected to have syntax error
+          git rm tests/Zend/Loader/_files/ParseError.php
+          git ls-files '**/*.php' | (xargs -l1 php -d display_errors=off -d log_errors=on -l && touch ok) | while read l; do echo -n .; done
```

are these ini settings really necessary? I could see the error in PHP 8.0 job was already outputted prior to the change from https://github.com/glensc/php-zf1s/commit/895f2d3e49c500447b2f3d53c1c87103f8d11296 - https://github.com/glensc/php-zf1s/runs/1516661591
```suggestion
          git ls-files '**/*.php' | (xargs -l1 php -l && touch ok) | while read l; do echo -n .; done
```

---

yes, because without them the errors are printed to stdout as well, and stdout is used for counting files (printing dots).

---

btw do you maybe know of any way to make it continue even on errors encountered, so that it would check all files and print all errors, instead of printing just the first one and bail? 🤔

---

it should test all files, I see no reason how it could fail, `xargs` invokes `php -l` for each file (`xargs -l`).

Altho I have not explicitly tested that is true because of the above reasons.

---

I checked https://github.com/glensc/php-zf1s/runs/1516737279?check_suite_focus=true and compared it with 7.4 job and noticed that there are less dots in the 8.0 job - meaning it most probably stops on first error

---

Indeed, looks like `xargs` behavior is to abort:

```
xargs: php: exited with status 255; aborting
```

I could not find an option from `xargs` manual page to change this behavior, so I'll use shell `while` loop instead.

---

I also found this https://github.com/StephaneBour/actions-php-lint/blob/7.4/entrypoint.sh - maybe this would be of any help.
Apparently there is an action ready to be used, but the author went for linking its version to php version used in `Dockerfile` and 8.0 is not available yet.

---

OK, looks better now:
- https://github.com/glensc/php-zf1s/runs/1523182249?check_suite_focus=true

---

Alternatives to using not our own code are listed in PR body and also in a note:
- https://github.com/zf1s/zf1/pull/37#issuecomment-740522629

The actions I've looked, had a specific one php version only, invoked docker, but we already have php in place. so these should be ran from composer, but I've planned composer setup in next PR as there are issues with php 8 to solve.

---

There's some optimization point too, add `-n` to skip loading ini files:
- https://github.com/phingofficial/phing/commit/8cfdd3ae293cb14c0e00816d6047d717153f6c9e

but it has a downside that error reporting needs to be reset or funny things happen with short open tag files. it's addressed in later commits in phing, but their trac being decommissioned it's harder to find those details.

---

Anyway, this is the latest one from this branch with two files with errors:
- https://github.com/glensc/php-zf1s/runs/1523341477?check_suite_focus=true

It's ready for merge from my side.

---

And job matching this branch:
- https://github.com/glensc/php-zf1s/runs/1523337914?check_suite_focus=true